diff --git a/lib/tls/CMakeLists.txt b/lib/tls/CMakeLists.txt
index b51fa3e77..45d022188 100644
--- a/lib/tls/CMakeLists.txt
+++ b/lib/tls/CMakeLists.txt
@@ -527,6 +527,7 @@ if (LWS_WITH_SSL)
 		endif()
 		if (NOT LWS_PLAT_FREERTOS AND OPENSSL_LIBRARIES)
 			list(INSERT LIB_LIST 0 ${OPENSSL_LIBRARIES})
+			list(APPEND LIB_LIST_AT_END ${OPENSSL_LIBRARIES})
 		endif()
 
 		if (NOT LWS_WITH_MBEDTLS AND NOT LWS_WITH_OPENHITLS)
