diff --git a/CMakeLists.txt b/CMakeLists.txt
index d246a59c..aaa2abf6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -542,7 +542,7 @@ if(HAS_LIBVORBIS)
 
     target_compile_options    (miniaudio_libvorbis PRIVATE ${COMPILE_OPTIONS})
     target_compile_definitions(miniaudio_libvorbis PRIVATE ${COMPILE_DEFINES})
-    target_link_libraries     (miniaudio_libvorbis PRIVATE libvorbis_interface)
+    target_link_libraries     (miniaudio_libvorbis PUBLIC miniaudio PRIVATE libvorbis_interface)
     target_include_directories(miniaudio_libvorbis PUBLIC  extras/decoders/libvorbis/)
 endif()
 
@@ -573,7 +573,7 @@ if(HAS_LIBOPUS)
 
     target_compile_options    (miniaudio_libopus PRIVATE ${COMPILE_OPTIONS})
     target_compile_definitions(miniaudio_libopus PRIVATE ${COMPILE_DEFINES})
-    target_link_libraries     (miniaudio_libopus PRIVATE libopus_interface)
+    target_link_libraries     (miniaudio_libopus PUBLIC miniaudio PRIVATE libopus_interface)
     target_include_directories(miniaudio_libopus PUBLIC  extras/decoders/libopus/)
 endif()
 
@@ -596,6 +596,7 @@ if (NOT MINIAUDIO_NO_EXTRA_NODES)
         target_include_directories(miniaudio_${name}_node PUBLIC  ${CMAKE_CURRENT_SOURCE_DIR}/extras/nodes/ma_${name}_node)
         target_compile_options    (miniaudio_${name}_node PRIVATE ${COMPILE_OPTIONS})
         target_compile_definitions(miniaudio_${name}_node PRIVATE ${COMPILE_DEFINES})
+        target_link_libraries     (miniaudio_${name}_node PUBLIC miniaudio)
 
         if(MINIAUDIO_BUILD_EXAMPLES)
             add_executable(miniaudio_${name}_node_example extras/nodes/ma_${name}_node/ma_${name}_node_example.c)
